apps/crud/src/app/loader.interceptor.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Observable, catchError, finalize, of, throwError } from 'rxjs';
+import { LoaderService } from './loader.service';
+import { Injectable } from '@angular/core';
+
+@Injectable({ providedIn: 'root' })
+export class LoaderInterceptor implements HttpInterceptor {
+  constructor(private loaderService: LoaderService) {}
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    this.loaderService.showLoader();
+    return next.handle(req).pipe(
+      finalize(() => {
+        this.loaderService.hideLoader();
+      }),
+      catchError((error) => {
+        this.loaderService.hideLoader();
+        console.log(error);
+        return of();
+      })
+    );
+  }
+}
```

Comment:
question: Did you implement it inside an entreprise application ? Personnaly, I always avoided it because it make the behavior too global. But that's a nice implementation 👍

Reply:
Previously I used this approach in my side project.
What could be a better approach other than state management?

Reply:
I mean, this is nice approach but all http request will trigger the loader, sometimes, you want it more local to a component to trigger a loader only on a row for instance, you don't want a global loader. With your approach, you don't have that flexibility.

Reply:
Ahh got it, as I am progressing with each step from the challenge. Step 2 suggests adding a global loader. Step 4 is expecting to implement state management and localised loader error. I will implement that in the next commit.